src/saveload/cargopacket_sl.cpp

```diff
@@ -17,6 +17,24 @@
 
 #include "../safeguards.h"
 
+/**
+ * Savegame conversion for version before cargo_movement was added.
+ */
+/* static */ void CargoPacket::AfterLoadBeforeCargoMovement()
+{
+	for (Station *st : Station::Iterate()) {
+		for (size_t i = 0; i < NUM_CARGO; i++) {
+			GoodsEntry *ge = &st->goods[i];
+			for (auto it = ge->cargo.Packets()->begin(); it != ge->cargo.Packets()->end(); it++) {
+				for (CargoPacket *cp : it->second) {
+					/* For cargo-movement, estimate it as the distance of the current station from the source. */
+					cp->cargo_movement = TileIndexToTileIndexDiffC(cp->source_xy, st->xy);
+				}
+			}
+		}
+	}
+}
```

Comment:
Reminder to self: dont cause desyncs, actually save the vector and restore in_vehicle. Silly goose.